test/test-functions

```diff
@@ -1706,14 +1706,20 @@ test_setup_cleanup() {
 }
 
 _test_cleanup() {
-    if mountpoint $TESTDIR/root; then
-        ddebug "umount $TESTDIR/root"
-        umount $TESTDIR/root
-    fi
-    if [[ $LOOPDEV && -b $LOOPDEV ]]; then
-        ddebug "losetup -d $LOOPDEV"
-        losetup -d $LOOPDEV
-    fi
+    # (post-test) cleanup should always ignore failure and cleanup as much as possible
+    (
+        set +e
+        if mountpoint $TESTDIR/root; then
+            ddebug "umount $TESTDIR/root"
+            umount $TESTDIR/root
+        fi
+        if [[ $LOOPDEV && -b $LOOPDEV ]]; then
+            ddebug "losetup -d $LOOPDEV"
+            losetup -d $LOOPDEV
+        fi
+        rm -fr "$TESTDIR"
+        rm -f "$STATEFILE"
+    ) || true
```

Comment:
Why both `+e` and `|| true`?

Reply:
Without the +e inside the subshell, the first failed cleanup step will exit the subshell, and the other cleanup steps won't happen.  The || true after the subshell prevents any cleanup errors from causing the main shell to exit.